GoogleUtilities/UserDefaults/GULUserDefaults.m

```diff
@@ -0,0 +1,236 @@
+// Copyright 2018 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "Private/GULUserDefaults.h"
+
+#import <GoogleUtilities/GULLogger.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSTimeInterval const kGULSynchronizeInterval = 1.0;
+
+static NSString *const kGULLogFormat = @"I-GUL%06ld";
+
+static GULLoggerService kGULLogUserDefaultsService = @"[GoogleUtilities/UserDefaults]";
+
+typedef NS_ENUM(NSInteger, GULUDMessageCode) {
+  GULUDMessageCodeInvalidKeyGet = 1,
+  GULUDMessageCodeInvalidKeySet = 2,
+  GULUDMessageCodeInvalidObjectSet = 3,
+  GULUDMessageCodeSynchronizeFailed = 4,
+  GULUDMessageCodeLibraryDirNotFound = 5,
```

Comment:
nit: It looks like this message code is never used. Not sure if this was used in a previous commit.

Reply:
Great catch! Fixed.